test/unit/vagrant/action/builtin/box_add_test.rb

```diff
@@ -84,6 +84,25 @@ def with_web_server(path, **opts)
     allow(box_collection).to receive(:find).and_return(nil)
   end
 
+  context "the download location is locked" do
+    let(:box_path) { iso_env.box2_file(:virtualbox) }
+    let(:mutex_path) {  env[:tmp_path].join("box" + Digest::SHA1.hexdigest("file://" + box_path.to_s) + ".lock").to_s }
+
+    it "raises a download error" do
+      # Lock file
+      f = File.open(mutex_path, "w+") 
+      f.flock(File::LOCK_EX|File::LOCK_NB) 
```

Comment:
I would have liked to put this in a before block like

```
before do
      f = File.open(mutex_path, "w+") 
      f.flock(File::LOCK_EX|File::LOCK_NB) 
    end
```

but then I don't have a good method for closing the file. Open to suggestions about a slicker way to do this 📖

Reply:
You could use an instance variable so it's still available for an `after` block. Something like:

```ruby
before do
    @f = File.open(mutex_path, "w+") 
    @f.flock(File::LOCK_EX|File::LOCK_NB) 
end

after { @f.close }

it ...
```